crates/doctor/src/manifest/trigger.rs

```diff
@@ -0,0 +1,236 @@
+use anyhow::{bail, ensure, Context, Result};
+use async_trait::async_trait;
+use toml::Value;
+use toml_edit::{Document, InlineTable, Item, Table};
+
+use crate::{Diagnose, Diagnosis, PatientApp, Treatment};
+
+use super::ManifestTreatment;
+
+/// TriggerDiagnosis detects problems with app trigger config.
+#[derive(Debug)]
+pub enum TriggerDiagnosis {
+    /// Missing app trigger section
+    MissingAppTrigger,
+    /// Invalid app trigger config
+    InvalidAppTrigger(&'static str),
+    /// HTTP trigger missing base field
+    HttpAppTriggerMissingBase,
+    /// HTTP component trigger missing route field
+    HttpComponentTriggerMissingRoute(String, bool),
+    /// Invalid HTTP component trigger config
+    InvalidHttpComponentTrigger(String, &'static str),
+}
+
+impl TriggerDiagnosis {
+    fn diagnose_app_trigger(config: Option<&Value>) -> Option<Self> {
+        let Some(config) = config else {
+            return Some(Self::MissingAppTrigger);
+        };
+        let Some(config) = config.as_table() else {
+            return Some(Self::InvalidAppTrigger("not a table"));
+        };
+        let Some(trigger_type) = config.get("type") else {
+            return Some(Self::InvalidAppTrigger("trigger table missing type"));
+        };
+        let Some(trigger_type) = trigger_type.as_str() else {
+            return Some(Self::InvalidAppTrigger("type must be a string"));
+        };
+        if trigger_type == "http" && config.get("base").is_none() {
+            return Some(Self::HttpAppTriggerMissingBase);
+        }
+        None
```

Comment:
If a manifest has multiple of these issues won't this implementation require them to keep running `spin doctor` until they have it fixed correctly?

Reply:
Got it, thanks.